dozer-types/src/json_types.rs

```diff
@@ -33,13 +79,105 @@ pub fn field_to_json_value(field: Field) -> Value {
         Field::Decimal(n) => Value::String(n.to_string()),
         Field::Timestamp(ts) => Value::String(ts.to_rfc3339_opts(SecondsFormat::Millis, true)),
         Field::Date(n) => Value::String(n.format(DATE_FORMAT).to_string()),
-        Field::Bson(b) => Value::from(b),
+        Field::Json(b) => json_value_to_serde_json(b),
         Field::Point(point) => convert_x_y_to_object(&point.0.x_y()),
         Field::Duration(d) => convert_duration_to_object(&d),
         Field::Null => Value::Null,
     }
 }
 
+pub fn json_value_to_serde_json(value: JsonValue) -> Value {
+    match value {
+        JsonValue::Null => Value::Null,
+        JsonValue::Bool(b) => Value::Bool(b),
+        JsonValue::Number(n) => Value::Number(Number::from_f64(*n).unwrap()),
```

Comment:
We should handle `NaN` and `Inf` instead of `unwrap`

Reply:
can't we just do `JsonValue::Number(n) => json!(*n),` then ?

Reply:
I guess it also panics. You can try it.

Reply:
```sql
insert into test_uuid values (
    'a0eebc99-9c0b-4ef8-bb6d-6bb9bd380a11',
    9,
    'Infinity'::jsonb,
    'Nan'::json,
    null,
    null
);
```

for `Infinity` -> Illegal floating point literal 
for `Nan` -> JSON standard does not allow such tokens

Reply:
> for Infinity -> Illegal floating point literal
> for Nan -> JSON standard does not allow such tokens

This is exactly how JSON works. What do you mean by "do we need it"?

Reply:
I guess `json!(*n)` panics if `n` is `NaN` of `Inf`. We need to handle this error instead of panic.

Reply:
no what i mean is there is no way that we are getting NaN of Inf as it's already considered illegal from db side

Reply:
The problem is `JsonValue::Number` is public, and it's very possible that it will be constructed in other places, in addition to in the connector. And that "other places" may produce `NaN` and put it in `JsonValue::Number`. This leaves us with three options:

1. Check if that `f64`  is finite and return error if not.
2. Don't check and panic.
3. Create a new type `FiniteF64` and use it instead of `f64` in `JsonValue::Number`.

Option 3 is best but also the most difficult. Option 2 is the easiest but the worst. Option 1 is just a way to avoid panicking but still doesn't go all the way into implementing a new type.

Reply:
sounds good - will proceed with option 3